crates/subsystem-util/subsystem-model-builder-util/src/builder/resolved_builder.rs

```diff
@@ -260,112 +260,129 @@ async fn resolve_module(
+
+                            fn create_diagnostic<T>(message: &str, span: Span, errors: &mut Vec<Diagnostic>,) -> Result<T, ModelBuildingError> {
+                                errors.push(
+                                    Diagnostic {
+                                        level: Level::Error,
+                                        message: message.to_string(),
+                                        code: Some("C000".to_string()),
+                                        spans: vec![SpanLabel {
+                                            span,
+                                            style: SpanStyle::Primary,
+                                            label: None,
+                                        }],
+                                    });
+                                Err(ModelBuildingError::Diagnosis(errors.clone()))
+                            }
+
+                            let kind_annot = match kind_annots.as_slice() {
+                                [] => {
+                                    create_diagnostic("Interceptor must have at least one of the before/after/around annotation", i.span, errors)
+                                }
+                                [single] => Ok(single),
+                                _ => create_diagnostic(
+                                    "Interceptor cannot have more than of the before/after/around annotations", i.span, errors
+                                ),
+                            }?;
+
+                            Result::<ResolvedInterceptor, ModelBuildingError>::Ok(ResolvedInterceptor {
+                                module_name: module.name.clone(),
+                                method_name: i.name.clone(),
+                                arguments: i
+                                    .arguments
+                                    .iter()
+                                    .map(|a| resolve_argument(a, types))
+                                    .collect(),
+                                interceptor_kind: kind_annot.clone(),
+                            })
+                        })
+                        .collect(),
+                        types_defined: module.types.iter().map(|m| m.name.clone()).collect(),
+                },
+            );
+        }
+        Some(AstAnnotationParams::None) => {
+            let script_path = module.base_exofile.clone();
+            process_script(module, base_system, &script_path)?;
+
+            resolved_modules.add(
+                &module.name,
+                ResolvedModule {
+                    name: module.name.clone(),
+                    script: vec![],
+                    script_path: "".to_string(),
+                    methods: vec![],
+                    interceptors: vec![],
+                    types_defined: module.types.iter().map(|m| m.name.clone()).collect(),
+                },
+            );
```

Comment:
I think this is why document_validator is failing, because is an empty module actually since is a @postgres module